### PR TITLE
Signup: Reset Verticals step state on signup complete

### PR DESCRIPTION
**Pull request by bisko**
A Signup Redux state reset mechanism was introduced in #8682 .

This PR brings the Survey Signup step to follow that pattern.

There shouldn't be any functional changes in the way Signup works. 

To test:

1. Checkout branch or use the Calypso.live link.
2. Start signup and make sure to use `surveyStepV2` AB variation.
3. Go through Signup
4. At the user step, before submitting, check IndexedDB if it contains the proper verticals you chose in `signup.steps.survey`
5. Submit the user form and wait for the `Continue` button to activate. Do NOT submit.
6. Follow the same steps from _4._ and verify that the `survey` state has been reset to an empty object - `{}`
7. Verify there were no console errors. 

cc @michaeldcain @coreh

### CODE REVIEW

File: client/state/signup/steps/survey/test/reducer.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+				otherText: 'test-other-text'
+			}
+		} ) ).to.be.eql( {
+			vertical: 'test-survey',
+			otherText: 'test-other-text'
+		} );
+	} );
+
+	it( 'should reset the survey on signup complete', () => {
+		expect( signupSurveyReducer( {}, {
```
Comment (coreh) [inline client/state/signup/steps/survey/test/reducer.js]: Shouldn't you pass here as the first arg:

```js
{
  vertical: 'test-survey',
  otherText: 'test-other-text'
}
```

To actually make sure the action is causing the reducer to erase the data?

File: client/state/signup/steps/survey/test/selectors.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			signup: {
+				steps: {
+					survey: {
+						vertical: 'test-survey',
+						otherText: 'test-other-text',
+					}
+				}
+			}
+		} ) ).to.be.eql( 'test-other-text' );
+	} );
```
Comment (coreh) [inline client/state/signup/steps/survey/test/selectors.js]: These tests are for your previous PR, right?

Comment (bisko) [inline client/state/signup/steps/survey/test/selectors.js]: Yes, just added them in this PR. Should I leave them here or split them as a separate PR?

File: client/state/signup/steps/survey/test/reducer.js (hunk 1 above)
Comment (bisko) [inline client/state/signup/steps/survey/test/reducer.js]: Thank you. Updated the code to make the test more meaningful! :)

File: client/state/signup/steps/survey/test/selectors.js (hunk 2 above)
Comment (coreh) [inline client/state/signup/steps/survey/test/selectors.js]: Nah, it's okay to leave them here IMO

File: client/state/signup/steps/survey/test/reducer.js (hunk 1 above)
Comment (coreh) [inline client/state/signup/steps/survey/test/reducer.js]: 👍

**Comment by coreh** [comment]
:shipit:
